src/custom-elements/src/components/tabs/tabs.tsx

```diff
@@ -0,0 +1,485 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  State,
+  h,
+  Listen
+} from '@stencil/core';
+import {
+  ACTIVE_CLASS,
+  DROPDOWN_CLASSES,
+  GENERIC_SIZES,
+  ICON_CLASS,
+  TABS_CLASSES,
+  UTILITY_CLASSES
+} from '../../constants/classes';
+import {
+  ANIMATION_DURATION,
+  TABS_SEE_MORE_DEFAULT_MESSAGE
+} from '../../constants/constants';
+import {
+  TabTrigger,
+  TabTriggerSizes,
+  TabTriggerPosition,
+  TabTriggerDirections
+} from '../../constants/types';
+import { ThreeStepsAnimation } from '../../utils/ThreeStepsAnimation';
+import { uuid4 } from '../../utils/utils';
+import _ from 'lodash';
+
+const MARGIN_LEFT = 24;
+
+@Component({
+  tag: 'chi-tabs',
+  styleUrl: 'tabs.scss',
+  scoped: true
+})
+export class Tabs {
+  @Element() el: HTMLElement;
+  /**
+   * To provide the data for Tabs
+   */
+  @Prop({ mutable: true, reflect: true }) tabs: TabTrigger[];
+  /**
+   *  To provide ID of the active tab
+   */
+  @Prop() activeTab = '';
+  /**
+   *  To enable animation of the border
+   */
+  @Prop() slidingBorder = false;
+  /**
+   *  To enable border
+   */
+  @Prop() border = false;
+  /**
+   *  To enable solid style of Tabs
+   */
+  @Prop() solid = false;
+  /**
+   *  To enable vertical variant of Tabs
+   */
+  @Prop() vertical = false;
+  /**
+   *  To configure See more Dropdown trigger message
+   */
+  @Prop() seeMoreMessage = TABS_SEE_MORE_DEFAULT_MESSAGE;
+
+  @State() sliding = false;
+  @State() isSeeMoreActive = false;
+  @State() isSeeMoreVisible = false;
+  @State() isSeeMoreTriggerMeasured = false;
+
+  /**
+   * Triggered when the user activates a tab
+   */
+  @Event({ eventName: 'chiTabChange' }) chiTabChange: EventEmitter<string>;
+
+  private activeTabElement: HTMLElement = null;
+  private animation: ThreeStepsAnimation;
+  private availableSpace = 0;
+  private dropdownKey = 0;
+  private slidingBorderElement: HTMLElement = null;
+  private lisArrayLength = 0;
+  private liSizes = {};
+  private seeMoreTriggerAnchorElement: HTMLAnchorElement;
+  private seeMoreTriggerElement: HTMLLIElement;
+  private seeMoreTriggerElementWidth = 0;
+  private ulElement: HTMLUListElement;
+  private uuid4: string = uuid4();
+  private seeMoreTriggerId: string = `see-more-trigger-${this.uuid4}`;
+  private slidingBorderLeft: any;
+  private slidingBorderWidth: any;
```

Comment:
Shouldn't these be strings?

Reply:
Numbers as the values are later concatenated `${slidingBorderLeft}px`